src/Modal.js

```diff
@@ -243,13 +251,26 @@ class Modal extends React.Component {
   }
 
   handleEscape(e) {
-    if (this.props.isOpen && this.props.keyboard && e.keyCode === 27 && this.props.toggle) {
+    if (this.props.isOpen && this.props.keyboard && e.keyCode === keyCodes.esc && this.props.toggle) {
       e.preventDefault();
       e.stopPropagation();
+
+      if (this.props.backdrop === 'static') {
+        this.handleStaticBackdropAnimation();
+        return;
+      }
+
       this.props.toggle(e);
     }
   }
 
+  handleStaticBackdropAnimation() {
+    this.setState({ showStaticBackdropAnimation: true });
+    setTimeout(() => {
```

Comment:
This is problematic if the component is unmounted before the timeout fires. It would cause `setState` to be executed on an unmounted component.
Maybe you can store the timeout on the component instance and clear it when the component unmounts.

Reply:
Thanks. I made changes to address this.